### PR TITLE
Set up a basic Github Action for CI

### PR DESCRIPTION
**Pull request by zackbrown**
@nbaksalyar this is my first time trying GH Actions, but perhaps this will do the trick re: #35 ?  

I also invited you to be a maintainer, so please feel free to review this PR!

### CODE REVIEW

**Comment by nbaksalyar** [comment]
Looking good! But if run on the workspace level, this unfortunately fails currently on Linux as it tries to build everything, including `pax-chassis-macos`, which doesn't build on non-Mac targets because of its specific dependencies.

Maybe we can start with simpler targets, like e.g. trying to build only `pax-example`? And then gradually introduce more and more checks & tests.

**Comment by zackbrown** [comment]
Once we merge `zb/dep-rethink`, this should be much better.  Part of the completed work over there is fixing the root workspace `cargo build`, as well as a bit of cleanup of the included/excluded workspace packages (chassis are removed from the workspace.) 

To reduce total work I'd recommend going ahead and merging this (if it seems functional, given a world where a root `cargo build`, `cargo test`, and `cargo check` are functional) and waiting for `zb/dep-rethink` to land for the satisfaction of green check marks.  If you'd prefer to wait to merge this, or to more deeply configure the GH action, I'd also support either.

**Review by nbaksalyar — APPROVED** [review]
Should be good to merge this now. 👍

**Pull request merged**
